src/sentry/static/sentry/app/components/group/externalIssueActions.tsx

```diff
@@ -37,87 +37,146 @@ class ExternalIssueActions extends AsyncComponent<Props, State> {
+    integration: GroupIntegration,
+    externalIssue: IntegrationExternalIssue
+  ) => {
+    let linked = JSON.parse(JSON.stringify(integration)) as GroupIntegration;
+    linked.externalIssues = [externalIssue];
+    this.setState(
+      {
+        linked: [...this.state.linked, linked],
+        unlinked: this.state.unlinked.filter(config => config.id !== linked.id),
+      },
+      () => this.closeModal()
+    );
+  };
 
+  renderBody() {
+    const {action, selectedIntegration, linked, unlinked} = this.state;
     return (
       <React.Fragment>
-        <IssueSyncListElement
-          onOpen={this.openModal}
-          externalIssueLink={issue ? issue.url : null}
-          externalIssueId={issue ? issue.id : null}
-          externalIssueKey={issue ? issue.key : null}
-          externalIssueDisplayName={issue ? issue.displayName : null}
-          onClose={this.deleteIssue.bind(this)}
-          integrationType={selectedIntegration.provider.key}
-          hoverCardHeader={t('Linked %s Integration', selectedIntegration.provider.name)}
-          hoverCardBody={
-            issue && issue.title ? (
-              <div>
-                <IssueTitle>{issue.title}</IssueTitle>
-                {issue.description && (
-                  <IssueDescription>{issue.description}</IssueDescription>
-                )}
-              </div>
-            ) : (
-              <IntegrationItem integration={selectedIntegration} />
-            )
-          }
-        />
+        {linked.length > 0 &&
+          linked.map(config => {
+            const {provider, externalIssues} = config;
+            let issue = externalIssues[0];
+            return (
+              <IssueSyncListElement
+                key={issue.id}
+                externalIssueLink={issue.url}
+                externalIssueId={issue.id}
+                externalIssueKey={issue.key}
+                externalIssueDisplayName={issue.displayName}
+                onClose={() => this.deleteIssue(config)}
+                integrationType={provider.key}
+                hoverCardHeader={t('Linked %s Integration', provider.name)}
+                hoverCardBody={
+                  <div>
+                    <IssueTitle>{issue.title}</IssueTitle>
+                    {issue.description && (
+                      <IssueDescription>{issue.description}</IssueDescription>
+                    )}
+                  </div>
+                }
+              />
+            );
+          })}
+
+        {unlinked.length > 0 && (
+          <IssueSyncListElement
+            externalIssueLink={null}
+            externalIssueId={null}
+            externalIssueKey={null}
+            externalIssueDisplayName={null}
+            integrationType={unlinked[0].provider.key}
+            hoverCardHeader={t('Linked %s Integration', unlinked[0].provider.name)}
+            hoverCardBody={
+              <Container>
+                {unlinked
+                  .sort((a, b) =>
```

Comment:
Added alphabetical sorting to prevent list reorders from multiple linking and unlinking actions.

Reply:
@NisanthanNanthakumar yea, if you had the getter I mentioned above it could handle the sorting as well.

Reply:
Any reason you don't want to put the sorting in `linkedIssuesFilter`?

Reply:
@scefali no, I can definitely add it in there!